frontend/packages/console-app/src/components/modals/restore-pvc/restore-pvc-modal.tsx

```diff
@@ -0,0 +1,242 @@
+import * as React from 'react';
+
+import { Form, FormGroup, Grid, GridItem, TextInput, Alert } from '@patternfly/react-core';
+import {
+  HandlePromiseProps,
+  ResourceIcon,
+  withHandlePromise,
+  validate,
+  resourceObjPath,
+  history,
+  RequestSizeInput,
+  Timestamp,
+} from '@console/internal/components/utils';
+import {
+  K8sResourceKind,
+  k8sCreate,
+  referenceFor,
+  VolumeSnapshotKind,
+  StorageClassResourceKind,
+  PersistentVolumeClaimKind,
+} from '@console/internal/module/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '@console/internal/components/factory';
+import {
+  NamespaceModel,
+  PersistentVolumeClaimModel,
+  VolumeSnapshotModel,
+  VolumeSnapshotClassModel,
+} from '@console/internal/models';
+import { getName, getNamespace, Status } from '@console/shared';
+import { StorageClassDropdown } from '@console/internal/components/utils/storage-class-dropdown';
+import { dropdownUnits } from '@console/internal/components/storage/shared';
+import { fetchK8s } from '@console/internal/graphql/client';
+
+import './restore-pvc-modal.scss';
+
+const RestorePVCModal = withHandlePromise((props: RestorePVCModalProps) => {
+  const { close, cancel, resource, errorMessage, inProgress, handlePromise } = props;
+  const [restorePVCName, setPVCName] = React.useState(`${getName(resource) || 'pvc'}-restore`);
+  const defaultSize: string[] = resource?.status?.restoreSize
+    ? validate.split(resource?.status?.restoreSize)
+    : [null, null];
+  const pvcRequestedSize = resource?.status?.restoreSize
+    ? `${defaultSize[0]} ${dropdownUnits[defaultSize[1]]}`
+    : '';
+  const [requestedSize, setRequestedSize] = React.useState(defaultSize?.[0] ?? '');
+  const [requestedUnit, setRequestedUnit] = React.useState(defaultSize?.[1] ?? 'Gi');
+  const [pvcSC, setPVCStorageClass] = React.useState('');
+  const [progress, setProgress] = React.useState(true);
+  const [driver, setDriver] = React.useState('');
+  const [pvcData, setPVCData] = React.useState<PersistentVolumeClaimKind>(null);
+  const namespace = getNamespace(resource);
+  const snapshotName = getName(resource);
+
+  React.useEffect(() => {
```

Comment:
you can just use `useK8sGet` hook which will simplify things quite a bit. Its not using GraphQL, but I dont think there's any performance boost in this case. Maybe we will change the hook to use GraphQL later.